### PR TITLE
Camera QR scanner

### PR DESCRIPTION
**Pull request by rkalman**
#1207
I understand the need of a general purpose scanner library, and recently html5-qrcode has gained 1D barcode functionality, however I have camera handling issues during testing. The zxing lib would be a great choice.

But if someone like to use a camera scanner with the internally generated QR codes only, maybe a lightweight solution like this is acceptable.

<a href="https://gitpod.io/#https://github.com/inventree/InvenTree/pull/2779"></a>

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -24,6 +24,7 @@ django-mptt==0.11.0             # Modified Preorder Tree Traversal
 django-redis>=5.0.0
 django-q==1.3.4                 # Background task scheduling
 django-sql-utils==0.5.0         # Advanced query annotation / aggregation
+django-sslserver==0.22          # Secure HTTP development server
```
Comment (matmair) [inline requirements.txt:27]: Why is this package added?

**Comment by matmair** [comment]
@rkalman ZXing went into maintenance mode so not relying on it would be nice. I took at the other library you suggested and it seems cool - what exactly did not work? Would you like me to take a look?

File: requirements.txt (hunk 1 above)
Comment (rkalman) [reply requirements.txt:27]: Pages have to be served over https in order to access the camera, so I needed a simple way to spin up a development server.

**Comment by rkalman** [comment]
I didn't spend too much time with it, but sometimes the camera stream didn't appear even on their demo page.
However it much more capable, but one function is missing:  detect whether a camera is present without asking permission from the user.

**Comment by matmair** [comment]
@SchrodingersGat we should maybe look at this

**Comment by SchrodingersGat** [comment]
@rkalman looks like the camera integration works "out of the box" (at least for me)! Very impressive. Are there any features you still want to integrate, or is this ready for review?

**Comment by rkalman** [comment]
Please review it, thank you.

**Comment by matmair** [comment]
@rkalman nice work!! Thank you

**Comment by matmair** [comment]
@schrodingersgat @rkalman if this merges (I hope so) there needs to be a license addition in the [credits section](https://inventree.readthedocs.io/en/latest/credits/)

File: InvenTree/templates/js/translated/barcode.js [hunk 2]
```diff
@@ -48,6 +52,44 @@ function makeBarcodeInput(placeholderText='', hintText='') {
...
+    $('#barcode_scan_video_container').show();
+    qrScanner.start();
+}
+
+function stopQrScanner() {
+    if (qrScanner != null) qrScanner.stop();
+    $('#barcode_scan_video_container').hide();
+}
+
+function onBarcodeScanClicked(e) {
```
Comment (matmair) [inline InvenTree/templates/js/translated/barcode.js:67]: @rkalman you need to export this function in the exported section - so just add 
`onBarcodeScanClicked`
in line 22 of this file

**Review by matmair — CHANGES_REQUESTED** [review]

**Comment by matmair** [comment]
@rkalman looks good now 👍

**Comment by rkalman** [comment]
Yes, thank you Mathias, I really had no idea why it is marked as unused...

**Comment by matmair** [comment]
@rkalman we missed updating the number of rendered script files. It should really run through now - there are no more tests that should/can fail.

**Comment by SchrodingersGat** [comment]
@rkalman thanks for this great addition :) I'm happy to merge this in now.

Can you please add a PR to the [documentation](https://github.com/inventree/inventree-docs/pulls) which adds:

- A short note to the [0.7.0 release notes](https://github.com/inventree/inventree-docs/blob/master/docs/releases/0.7.0.md) referencing this PR and feature description
- Update the "frontend libraries" section of the [credits](https://github.com/inventree/inventree-docs/blob/master/docs/credits.md) page

**Pull request merged**